index.js

```diff
@@ -0,0 +1,258 @@
+        .get()
+        .then(users => {
+            return response
+                .status(200)
+                .json(users);
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "Could not find list of users." })
+        });
+});
+
+server.get('/api/users/:id', (request, response) => {
+    const id = request.params.id;
+
+    userDb
+        .get(id)
+        .then(user => {
+            if (!user) {
+                return response
+                    .status(404)
+                    .json({ Error: "Could not find user." })
+            } else return response
+                .status(200)
+                .json(user);
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "User info could not be retrieved." })
+        });
+});
+
+server.post('/api/users', toCaps, (request, response) => {
+    const name = request.name;
+    const newUser = { name };
+
+    if (!newUser.name) {
+        return response
+            .status(400)
+            .send({ Error: "Please enter a name for the user" });
+    } else if (newUser.name.length > 128) {
+        return response
+            .status(400)
+            .send({ Error: "User name must be 128 or less characters" });
+    }
+
+    userDb
+        .insert(newUser)
+        .then(userID => {
+            const { id } = userID;
+            userDb
+                .get(id)
+                .then(user => {
+                    return response
+                        .status(201)
+                        .json(user);
+                });
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "There was an error while saving the user" })
+        });
+});
+
+server.put('/api/users/:id', toCaps, (request, response) => {
+    const id = request.params.id;
+    const name = request.name;
+    const updatedUser = { name };
+
+    if (!id) {
+        return response
+            .status(404)
+            .send({ Error: `User with the following ID does not exist: ${id}` });
+    } else if (!updatedUser.name) {
+        return response
+            .status(400)
+            .send({ Error: "Please enter a name for the user" });
+    }
+
+    userDb
+        .update(id, updatedUser)
+        .then(user => {
+            return response
+                .status(200)
+                .json(user);
+        })
+        .catch(() => {
+            return response
+                .status(500)
+                .json({ Error: "The user info could not be modified" })
+        });
+});
+
+server.delete('/api/users/:id', (request, response) => {
+    const id = request.params.id;
+
+    if (!id) {
+        return response
+            .status(404)
+            .json({ Error: `There is no user with the following ID: ${id}` })
+    }
+
+    userDb
+        .remove(id)
+        .then(removedUser => {
```

Comment:
How can you handle the case where the given id was not found in the db to be removed?